### PR TITLE
List structure

### PR DESCRIPTION
**Pull request by Maclenn77**
Hi, Microverse team!

I'm submitting this project for creating a minimalist TODO list.

- I set up the project with webpack, based on the previous webpack exercise.
- I created an index.html file and write the HTML markup there. 
- I created an empty To Do List placeholder (`<div>` or `<ul>` element). 
- The index.html file was set as a template using the HTML Webpack Plugin.
- I created an index.js file and set an array of some simple to-do tasks (array of objects). 
- Each task object contains three keys: description [string], completed [bool], index: [number].
- I wrote a function to iterate over the tasks array and populate an HTML list item element for each task.
- The dynamically created list of tasks is displayed in the dedicated placeholder. 
- The list appears in order of the index values for each task.
- I created a style.css and set rules for the To-Do List. 
- CSS is loaded by Webpack Style/CSS Loader. 
- The list is intended to be a clone of the part of the minimalist project used as an example.

Have a good day! :smiley_cat:

### CODE REVIEW

**Comment by omar-labana** [comment]
Hi @Maclenn77 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project where he node_modules folder is found in the repo.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Maclenn77** [comment]
> 
> You have submitted a project where he node_modules folder is found in the repo.
> 

Hi, @omar-labana. I added node_modules to .gitignore

I don't understand what I did wrong and why it's appearing. My gitignore file includes node_modules.

**Comment by Maclenn77** [comment]
I submitted for review again in the wrong activity and I was not able to unsubmit.

**Comment by omar-labana** [comment]
Hi <STUDENT_MENTION>, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted the wrong activity, I can only see the basic list for the previous project milestone.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    return 'checked';
+  }
+  return '';
+}
+
+function displayTasks() {
+  for (let i = 0; i < toDoList.length; i += 1) {
+    tasksBoard.insertAdjacentHTML('beforeend', `<li><input type='checkbox' ${isChecked(toDoList[i].completed)}>${toDoList[i].description}</li>`);
+  }
+}
```
Comment (D4nielJ) [inline src/index.js]: - [x] Let's make sure that our displayTasks function renders the tasks by the order of the index. This way, if the task are rearranged in the future, it does not matter in which place of the array they are, they're going to get rendered in order.

Imagine that the user moves the tasks around, the toDoList array will remain `[task1, task2, task3]`, but the index of the tasks might be something different.

For this you can use the `sort` function like this: `sortedTasks = toDoList.sort((a, b) => a - b)` and then render from there.

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+module.exports = {
+  entry: {
```
Comment (D4nielJ) [inline webpack.config.js:6]: [OPTIONAL] I noticed that when I start the development server with `npm run start` a notice complaining that webpack does not have the mode set came up. You can fix this by setting up the `mode` to `'development'` in this file.

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @Maclenn77, 

You're doing great! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Pérez, 
## Congratulations 🎉 Thank you for implementing changes suggested previously 🙏

## Happy Coding 👏👏👏
#### Please leave any questions or comments in the PR if you have any doubts or unclarity!
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**